v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/schema/typemapping/provider/PostgreSQLMappingProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.schema.typemapping.provider;
+
+import com.google.cloud.teleport.v2.source.reader.io.schema.typemapping.UnifiedTypeMapping;
+import com.google.cloud.teleport.v2.source.reader.io.schema.typemapping.provider.unified.UnifiedMappingProvider;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Provides a set of {@link org.apache.avro.Schema Avro Schemas} that each of the PostgreSQL
+ * database's type must map into.
+ *
+ * @see <a href="https://cloud.google.com/datastream/docs/unified-types#map-psql">Mappings of
+ *     unified types to source and destination data types</a>
+ */
+public final class PostgreSQLMappingProvider {
+  // TODO(thiagotnunes): Add missing type mappings
+  private static final ImmutableMap<String, UnifiedTypeMapping> MAPPING =
+      ImmutableMap.<String, UnifiedMappingProvider.Type>builder()
+          .put("BIGINT", UnifiedMappingProvider.Type.LONG)
+          .put("CHARACTER VARYING", UnifiedMappingProvider.Type.STRING)
+          .put("DATE", UnifiedMappingProvider.Type.TIMESTAMP)
```

Comment:
We follow Datastream's  [unified mapping](https://cloud.google.com/datastream/docs/unified-types#map-psql) to maintain compatibility and code reusability with live template.
I think PG data map's to date logical type.

For MySQL this maps to TimeStamp but I am not sure why it's done that way :) (outside of historic reasons)

Reply:
Updated